InnerEye/ML/configs/segmentation/BasicModel2Epochs.py

```diff
@@ -42,7 +42,10 @@ def __init__(self, **kwargs: Any) -> None:
             num_epochs=2,
             recovery_checkpoint_save_interval=1,
             use_mixed_precision=True,
-            azure_dataset_id=AZURE_DATASET_ID,
+            # azure_dataset_id=AZURE_DATASET_ID,
```

Comment:
I made a copy of the dataset in `innereyepubliccheckpoint > datasets > innereyepublicdatasets/datasets/2339eba2-8ec5-4ccb-86ff-c170470ac6e2_geonorm_with_train_test_split_2020_05_26` but with all the patients other than 0 to 9 deleted (i.e. their image folders and their rows in dataset.csv deleted) because patients 0-9 are the only ones in the data splits for the model `BasicModel2EpochsMoreData`.

To make this partial ground truth I additionally removed the row `5	5/spinalcord.nii.gz	spinalcord	402ba5d42f37357f18f29af17b0846cbca4c430fea5b15a6baad5ec29dd6c9ba	b7f757fb-12e0-489e-a6da-f64895cdd229` and deleted the file `5/spinalcord.nii.gz`.

The command was 
    > python InnerEye/ML/runner.py --allow_incomplete_labels --train=False --azureml --model=BasicModel2EpochsMoreData --run_recovery_id=main_1625070510_eec68070

It is also important to include this edit of `BasicModel2Epochs.py` so that it picks up the new azure_dataset_id,
which for me was `geonorm_with_train_test_split_partial_test_data_2021_06_30`

After one bug fix ([here](https://github.com/microsoft/InnerEye-DeepLearning/blob/b6241d24b61c1590f045a4799d991bd9b0bb4370/InnerEye/ML/lightning_base.py#L165-L172)) the end-to-end test ran and the output CSVs and reports included patient counts reflecting the incomplete ground truth as expected.

Reply:
I have also added a unit test ([here](https://github.com/microsoft/InnerEye-DeepLearning/blob/c2f0d668e8cefc1c3868de86c62131f38ca2531d/Tests/ML/test_lightning_containers.py#L286-L314)) to cover that bug.